### ISSUE TITLE
Feature Request: Duplicate row option

### ISSUE THREAD

**Issue by utkarshcmu** [grafana/grafana#2910]
<img width="797" alt="screen shot 2015-10-09 at 3 18 54 am" src="https://cloud.githubusercontent.com/assets/1019609/10380934/aaef34b6-6e34-11e5-848b-22e440c67343.png">

This should just duplicate the selected row and another row just below it with all the contained panels.

**Comment by utkarshcmu** [comment]
Somewhat related to #1004

**Comment by kawikao** [comment]
+1. Either this or disable panel resizing when dragging to another row.

**Comment by gdubicki** [comment]
I am totally surprised that this is not requested more. The most common way I develop dashboards is to create and polish a single row, then copy it a few times and modify every copy a bit for different metrics. And to do that for a single graph row I currently have to do 4 operations: 1. add a new row, 2. duplicate the source graph, 3. move the new graph to the new row, 4. extend the source graph again to 100% width as it was shrunk to 50% after step 2. That's a lot of clicking... :(

**Comment by kawikao** [comment]
I guess copying a nested JSON object and generating random row and panel IDs is more complicated than I thought.

**Comment by chow1186** [comment]
Hi utkarshcmu,
I don't see the duplicate row option in Grafana. which version is that screenshot you posted? do we have to enable "duplicate row" option or is it a default option?

**Comment by gdubicki** [comment]
It's from @utkarshcmu 's Pull Request (https://github.com/grafana/grafana/pull/7133) which was rejected, @chow1186 . :(

**Comment by torkelo** [comment]
closing this, in v5 we still have rows (mainly for backward compatibility  but they are not as crucial anymore so not sure we need this.

**Comment by mammuthus** [comment]
Hello. Started new issue but found the same:)

Is this a final decision? Rows are ery usefull and If you want to make full row duplicate with it's structure and graphs, It will be good to have such button.
![image](https://user-images.githubusercontent.com/1700286/49066578-a2e4f280-f232-11e8-903e-615924c02955.png)
